lib/portage/tests/resolver/test_runtime_cycle_merge_order.py

```diff
@@ -74,3 +74,147 @@ def testRuntimeCycleMergeOrder(self):
                 self.assertEqual(test_case.test_success, True, test_case.fail_msg)
         finally:
             playground.cleanup()
+
+    def testBuildtimeRuntimeCycleMergeOrder(self):
```

Comment:
I might be able to tighten this up or add a shorter test case but I'll look at that later.